tests/core/pyspec/eth2spec/test/eip7732/epoch_processing/test_process_builder_pending_payments.py

```diff
@@ -0,0 +1,301 @@
+    withdrawal = state.builder_pending_withdrawals[len(state.builder_pending_withdrawals) - 1]
+    assert withdrawal.fee_recipient == fee_recipient
+    assert withdrawal.amount == amount
+    assert withdrawal.builder_index == builder_index
+    assert withdrawal.withdrawable_epoch > spec.get_current_epoch(state)
+
+    # Payment should be rotated out
+    assert state.builder_pending_payments[0].weight == 0
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_multiple_above_quorum(spec, state):
+    """Test multiple payments above quorum threshold."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    quorum = spec.get_builder_payment_quorum_threshold(state)
+    weight = quorum + 1
+
+    # Add multiple payments above quorum
+    num_payments = min(3, spec.SLOTS_PER_EPOCH)
+    for i in range(num_payments):
+        builder_index = i
+        amount = spec.MIN_ACTIVATION_BALANCE + i * spec.EFFECTIVE_BALANCE_INCREMENT
+        fee_recipient = bytes([0x10 + i]) + b"\x00" * 19
+
+        payment = create_builder_pending_payment(spec, builder_index, amount, weight, fee_recipient)
+        state.builder_pending_payments[i] = payment
+
+    pre_builder_pending_withdrawals = len(state.builder_pending_withdrawals)
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # All payments should result in withdrawals
+    assert len(state.builder_pending_withdrawals) == pre_builder_pending_withdrawals + num_payments
+
+    # Check each withdrawal
+    for i in range(num_payments):
+        withdrawal = state.builder_pending_withdrawals[pre_builder_pending_withdrawals + i]
+        expected_amount = spec.MIN_ACTIVATION_BALANCE + i * spec.EFFECTIVE_BALANCE_INCREMENT
+        expected_fee_recipient = bytes([0x10 + i]) + b"\x00" * 19
+
+        assert withdrawal.amount == expected_amount
+        assert withdrawal.builder_index == i
+        assert withdrawal.fee_recipient == expected_fee_recipient
+        assert withdrawal.withdrawable_epoch > spec.get_current_epoch(state)
+
+    # All payments should be cleared (weights set to 0)
+    for i in range(num_payments):
+        assert state.builder_pending_payments[i].weight == 0
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_mixed_weights(spec, state):
+    """Test mix of payments above and below quorum."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    quorum = spec.get_builder_payment_quorum_threshold(state)
+
+    # Add payments with different weights
+    payments_data = [
+        (0, spec.MIN_ACTIVATION_BALANCE, quorum + 1),  # Above threshold
+        (1, spec.MIN_ACTIVATION_BALANCE, quorum // 2),  # Below threshold
+        (2, spec.MIN_ACTIVATION_BALANCE, quorum + 100),  # Above threshold
+        (3, spec.MIN_ACTIVATION_BALANCE, quorum - 1),  # Below threshold
+    ]
```

Comment:
Can we order these least to greatest weight? And add one where it's exactly `quorum`?